tasks/set-plugin-list.yml

```diff
@@ -0,0 +1,56 @@
+- name: "Setting fact for juseppe_plugins"
+  set_fact:
+    juseppe_plugins: []
+
+- name: "Setting fact for juseppe_plugins -> wizard plugins"
+  set_fact:
+      juseppe_plugins: "{{ cj_wizard_plugins }}"
+  when: (cj_wizard is defined ) and (cj_wizard | bool)
+
+- name: "merge plugin lists -> jenkins theme"
+  set_fact:
+    juseppe_plugins: "{{ juseppe_plugins + cj_theme_plugins }}"
+  when: (cj_theme is defined) and (cj_theme | bool)
+
+- name: "merge plugin lists -> jenkins auth strategy"
+  set_fact:
+    juseppe_plugins: "{{ juseppe_plugins + cj_setup_matrix_auth_strategy_plugins }}"
+  when: (cj_setup_matrix_auth_strategy is defined) and (cj_setup_matrix_auth_strategy | bool)
+
+- name: "merge plugin lists -> jenkins DSL plugin"
+  set_fact:
+    juseppe_plugins: "{{ juseppe_plugins + cj_dsl_plugins }}"
+  when: (cj_setup_dsl_plugins is defined) and (cj_setup_dsl_plugins | bool)
+
+- block:
+  - name: "Setting fact for juseppe_plugins -> base"
+    set_fact:
+      juseppe_plugins: "{{ cj_base_plugins }}"
+    when: (cj_base is defined) and (cj_base | bool)
+
+  - name: "merge plugin lists -> base + goodies"
+    set_fact:
+      juseppe_plugins: "{{ juseppe_plugins }} + {{ cj_goodie_plugins }}"
+    when: (cj_goodies is defined) and (cj_goodies | bool)
+
+  - name: "merge plugin lists -> github"
+    set_fact:
+      juseppe_plugins: "{{ juseppe_plugins }} + {{ cj_github_plugins }}"
+    when: (cj_github is defined) and (cj_github | bool)
+
+  - name: "merge plugin lists -> pipeline"
+    set_fact:
+      juseppe_plugins: "{{ juseppe_plugins }} + {{ cj_pipeline_plugins }}"
+    when: (cj_pipeline is defined) and (cj_pipeline | bool)
+
+  - name: "merge plugin lists -> blueocean"
+    set_fact:
+      juseppe_plugins: "{{ juseppe_plugins }} + {{ cj_blueocean_plugins }}"
+    when: (cj_blueocean is defined) and (cj_blueocean | bool)
+
+  - name: "merge plugin lists -> workflow"
+    set_fact:
+      juseppe_plugins: "{{ juseppe_plugins }} + {{ cj_workflow_plugins }}"
+    when: (cj_workflow is defined) and (cj_workflow | bool)
+
+  when: (cj_wizard is not defined ) and (cj_wizard | bool)
```

Comment:
I tried to find a short resolution to merge dicts, but got into corners so I just used the one from jenkins.